File: app-expand/src/main/java/com/zxn/expandapp/widget/ExpandListView.java
package com.zxn.expandapp.widget;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.ListView;

/**
 * ExpandListView
 * 一个可以和ScrollView嵌套的ListView,解决高度问题.
 */
public class ExpandListView extends ListView {
	public ExpandListView(Context context) {
		super(context);
	}

	public ExpandListView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public ExpandListView(Context context, AttributeSet attrs,
						  int defStyle) {
		super(context, attrs, defStyle);
	}

	@Override
	protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
		int expandSpec = MeasureSpec.makeMeasureSpec(Integer.MAX_VALUE >> 2,
				MeasureSpec.AT_MOST);
		super.onMeasure(widthMeasureSpec, expandSpec);
	}
}

File: expand-widget/src/main/java/com/zxn/widget/ExpandListView.kt
package com.zxn.widget

import android.content.Context
import android.util.AttributeSet
import android.widget.ListView

/**一个可以和ScrollView嵌套的ListView,解决高度问题.
 *  Updated by zxn on 2020.10.17.
 */
class ExpandListView : ListView {
    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?,
                defStyle: Int) : super(context, attrs, defStyle) {
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val expandSpec = MeasureSpec.makeMeasureSpec(Int.MAX_VALUE shr 2,
                MeasureSpec.AT_MOST)
        super.onMeasure(widthMeasureSpec, expandSpec)
    }
}